packages/api/amplify_api/lib/amplify_api.dart

```diff
@@ -36,13 +37,8 @@ abstract class AmplifyAPI extends APIPluginInterface {
     List<APIAuthProvider> authProviders = const [],
     ModelProviderInterface? modelProvider,
   }) {
-    if (zIsWeb || Platform.isWindows || Platform.isMacOS || Platform.isLinux) {
-      throw UnsupportedError('This platform is not supported yet');
-    }
-    return AmplifyAPIMethodChannel(
-      authProviders: authProviders,
-      modelProvider: modelProvider,
-    );
+    return AmplifyAPIDart(
+        authProviders: authProviders, modelProvider: modelProvider);
```

Comment:
Since all this factory method does is redirect now, you can replace it with

```dart
factory AmplifyAPI(...) = AmplifyAPIDart;
```

Reply:
I took this suggestion.

Can you double check that the syntax I used is valid with the parameters? Do the parameters get passed on implicitly?